### DISCUSSION TITLE
Several packages for the plugin

### DISCUSSION THREAD

**Discussion by ivanpopelyshev** [pixijs/pixijs#7398] [Ideas]
Oh great keeper of packages, @bigtimebuddy !
Respected master of build tools, @SukantPal !

I have a question for you two and maybe for others in community.

### Situation

As you see here: https://github.com/pixijs/pixi-spine/tree/next
New version of pixi-spine for v6 is separated into multiple modules:

1. `@pixi-spine/base` holds universal stuff across all spine plugins (core) + `SpineBase` class that is actual integration with pixi
2. `@pixi-spine/runtime-3.8` holds runtime for spine 3.8, reworked `core` from https://github.com/EsotericSoftware/spine-runtimes/tree/3.8/spine-ts and factory class `Spine` that uses corresponding skeleton.
3. `@pixi-spine/loader-3.8` is a pinacle of hacks that has to be used now and buried deep later. pixi loader is not suitable for multi-file resources, its pain to support all ways people figured out. That's why it has separate package - i'll make new loader with other name in future.
4. `@pixi-spine` re-exports everything. In future it will contain all runtimes and loaders so people wont think "why my 4.0-beta model cant be loaded here, at the same time as 3.7 model from old demo"

customization of build will look like pixi-customize, however there will be conflicts, like "dont include `loader-3.8` at the same time as `loader-3.7`, use `uni-loader` for that"

Also, I will move some stuff from `pixi-heaven` regarding spine - double-tint feature.

In the end, number of packages there will reach ~10. 

### Question

Should I name packages `@pixi-spine/xxx` or `@pixi/spine-xxx` or `@pixi/spine/xxx` ?

**Comment by andrewstart** [comment]
I'd go with `@pixi/spine-xxx`.

**Reply by ivanpopelyshev** [reply]
In this case, if we add it to https://pixijs.io/customize/ it will be whole group, like Filters or Canvas

**Comment by ivanpopelyshev** [comment]
What if I leave main bundle as `pixi-spine` ? It has history, like `pixi.js`

**Comment by bigtimebuddy** [comment]
Does your main bundle include just latest spine runtime, or all runtimes?

In general, my preference is to use `@pixi-spine` namespace if you're going to have about a dozen packages so it does not pollute `@pixi` scope (I have already reserved this org on npm for you) So I'm in favor of `@pixi-spine/runtime-3.7`, `@pixi-spine/base`, etc. 

For the main bundle, if you decided to keep `pixi-spine` or use `@pixi/spine` (my preference) either are fine with me.

**Reply by ivanpopelyshev** [reply]
That's what I wanted to hear :) Thanks!
